src/andromede/study/resolve_components.py

```diff
@@ -36,48 +36,51 @@
     dataframe_to_time_series,
     load_ts_from_txt,
 )
-from andromede.study.parsing import InputComponent, InputPortConnections, InputStudy
+from andromede.study.parsing import InputComponent, InputPortConnections, InputSystem
 
 
 @dataclass(frozen=True)
-class NetworkComponents:
+class System:
     components: Dict[str, Component]
     nodes: Dict[str, Component]
     connections: List[PortsConnection]
 
 
-def network_components(
+def system(
     components_list: Iterable[Component],
     nodes: Iterable[Component],
     connections: Iterable[PortsConnection],
-) -> NetworkComponents:
-    return NetworkComponents(
+) -> System:
+    return System(
         components=dict((m.id, m) for m in components_list),
         nodes=dict((n.id, n) for n in nodes),
         connections=list(connections),
     )
 
 
-def resolve_components_and_cnx(
-    input_comp: InputStudy, library: Library
-) -> NetworkComponents:
+def resolve_system(input_system: InputSystem, libraries: dict[str, Library]) -> System:
     """
     Resolves:
     - components to be used for study
     - connections between components"""
-    components_list = [_resolve_component(library, m) for m in input_comp.components]
-    nodes = [_resolve_component(library, n) for n in input_comp.nodes]
+    components_list = [
+        _resolve_component(libraries, m) for m in input_system.components
+    ]
+    nodes = [_resolve_component(libraries, n) for n in input_system.nodes]
     all_components: List[Component] = components_list + nodes
     connections = []
-    for cnx in input_comp.connections:
+    for cnx in input_system.connections:
         resolved_cnx = _resolve_connections(cnx, all_components)
         connections.append(resolved_cnx)
 
-    return network_components(components_list, nodes, connections)
+    return system(components_list, nodes, connections)
 
 
-def _resolve_component(library: Library, component: InputComponent) -> Component:
-    model = library.models[component.model]
+def _resolve_component(
+    libraries: dict[str, Library], component: InputComponent
+) -> Component:
+    lib_id, model_id = component.model.split(".")
```

Comment:
Should we accept multiple levels of lib, and only split on the last dot ?
like `rte.drd.model-of-the-future` ?
If not we should check that there is indeed only one dot.
Optionally this could be delegated to a container of libraries, see proposition above.

Reply:
Like the previous comment, I let the reflexions on this question for the industrial part, the Python code will just follow C++ choices